From e8f7bc573c88efd6577ad915647206329de449ae Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:17:13 -0400
Subject: [alsa] hda: Fix missing VREF setup for Mac Pro 1,1

Message-id: <140972823573-84-patchset-jkysela@redhat.com>
Patchwork-id: 90784
O-Subject: [RHEL 7.1 PATCH 083/442] ALSA: hda - Fix missing VREF setup for Mac Pro 1,1
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

Mac Pro 1,1 with ALC889A codec needs the VREF setup on NID 0x18 to
VREF50, in order to make the speaker working.  The same fixup was
already needed for MacBook Air 1,1, so we can reuse it.

Reported-by: Nicolai Beuermann <mail@nico-beuermann.de>
Cc: <stable@vger.kernel.org>
Signed-off-by: Takashi Iwai <tiwai@suse.de>

(picked from kernel: c20f31ec421ea4fabea5e95a6afd46c5f41e5599)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/sound/pci/hda/patch_realtek.c b/sound/pci/hda/patch_realtek.c
index cbeb67d..af7a0c3 100644
--- a/sound/pci/hda/patch_realtek.c
+++ b/sound/pci/hda/patch_realtek.c
@@ -1821,6 +1821,7 @@ enum {
  ALC889_FIXUP_IMAC91_VREF,
  ALC889_FIXUP_MBA11_VREF,
  ALC889_FIXUP_MBA21_VREF,
+ ALC889_FIXUP_MP11_VREF,
  ALC882_FIXUP_INV_DMIC,
  ALC882_FIXUP_NO_PRIMARY_HP,
  ALC887_FIXUP_ASUS_BASS,
@@ -2190,6 +2191,12 @@ static const struct hda_fixup alc882_fixups[] = {
   .chained = true,
   .chain_id = ALC889_FIXUP_MBP_VREF,
  },
+ [ALC889_FIXUP_MP11_VREF] = {
+  .type = HDA_FIXUP_FUNC,
+  .v.func = alc889_fixup_mba11_vref,
+  .chained = true,
+  .chain_id = ALC885_FIXUP_MACPRO_GPIO,
+ },
  [ALC882_FIXUP_INV_DMIC] = {
   .type = HDA_FIXUP_FUNC,
   .v.func = alc_fixup_inv_dmic_0x12,
@@ -2253,7 +2260,7 @@ static const struct snd_pci_quirk alc882_fixup_tbl[] = {
  SND_PCI_QUIRK(0x106b, 0x00a0, "MacBookPro 3,1", ALC889_FIXUP_MBP_VREF),
  SND_PCI_QUIRK(0x106b, 0x00a1, "Macbook", ALC889_FIXUP_MBP_VREF),
  SND_PCI_QUIRK(0x106b, 0x00a4, "MacbookPro 4,1", ALC889_FIXUP_MBP_VREF),
- SND_PCI_QUIRK(0x106b, 0x0c00, "Mac Pro", ALC885_FIXUP_MACPRO_GPIO),
+ SND_PCI_QUIRK(0x106b, 0x0c00, "Mac Pro", ALC889_FIXUP_MP11_VREF),
  SND_PCI_QUIRK(0x106b, 0x1000, "iMac 24", ALC885_FIXUP_MACPRO_GPIO),
  SND_PCI_QUIRK(0x106b, 0x2800, "AppleTV", ALC885_FIXUP_MACPRO_GPIO),
  SND_PCI_QUIRK(0x106b, 0x2c00, "MacbookPro rev3", ALC889_FIXUP_MBP_VREF),
-- 
1.7.1